tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressions.cs

```diff
@@ -137,5 +137,21 @@ public void Unsupported_bson_types()
         {
             _ = GetMongoQueryable<ClassWithBsonTypes>().Where(o => o.BsonDocument.ElementCount == 10);
         }
+
+        [InvalidLinq("{document}.Quantity is not supported.")]
+        [InvalidLinq("{document}._AppleID is not supported.")]
+        public void Warnings_due_to_bson_ignore()
+        {
+            _ = GetMongoQueryable<Fruit>().Where(f => f.Quantity == 22);
+            _ = GetMongoQueryable<Apple>().Where(a => a._AppleID == "Apple ID");
+        }
+
+        [InvalidLinq("Class System.Int32 cannot be assigned to Class MongoDB.Analyzer.Helpers.Linq.GenType_Class_16.  Ensure that known types are derived from the mapped class. (Parameter 'type')")]
```

Comment:
We probably need types names mapping replacement in exceptions as well (`GenType_Class_16`) .
Can you see if it's easy to do in this ticket, or split to another.

Reply:
I was able to do it in this ticket using Regex. Similar logic to how ConstantsMapper replaces constants